python/ray/tune/execution/experiment_state.py

```diff
@@ -182,107 +166,86 @@ def checkpoint(
-
-        # Finish
-        self._last_save_time = time.time()
-
-    def sync_up(self, force: bool = False, wait: bool = False) -> bool:
-        syncer = self._storage.syncer
-
-        if not syncer:
-            return False
-
-        # Always exclude checkpoints in the new persistence path.
-        # TODO(justinvyu, krfricke): Ideally, this excludes all trial directories.
-        # But for now, this is needed to upload driver artifacts that live in the
-        # trial directory.
-        exclude = _DRIVER_SYNC_EXCLUDE_PATTERNS
-        experiment_local_path = self._storage.experiment_driver_staging_path
-        experiment_fs_path = self._storage.experiment_fs_path
+        def wait_for_sync():
+            try:
+                self._storage.syncer.wait()
+            except Exception:
+                logger.error(
+                    "Saving experiment state to storage at "
+                    f"'{self._storage.experiment_fs_path}' failed with exception: ",
+                    exc_info=True,
+                )
 
         if force:
-            # Wait until previous sync command finished
-            try:
-                syncer.wait()
-            except TimeoutError as e:
+            start_time = time.monotonic()
+            wait_for_sync()
+            wait_time = time.monotonic() - start_time
+            if wait_time > self._slow_sync_threshold:
                 logger.warning(
-                    "The previous sync of the experiment directory to the cloud "
-                    f"timed out with the error: {str(e)}\nSyncing will be retried. "
-                    + _EXPERIMENT_SYNC_TIMEOUT_MESSAGE
+                    "Saving the experiment state (which holds a global view "
+                    "of trial statuses and is used to restore the experiment) "
+                    f"took ~{wait_time:.2f} seconds, which may be a performance "
+                    "bottleneck.\n"
+                    f"{_SLOW_SYNC_WARNING.format(threshold=self._slow_sync_threshold)}"
                 )
-            except Exception as e:
+
+        time_since_last_sync = (
+            time.monotonic() - self._last_sync_time
+            if self._last_sync_time is not None
+            else None
+        )
+        launched_sync = self._storage.syncer.sync_up(
+            driver_staging_path, self._storage.experiment_fs_path
+        )
+        if launched_sync:
+            if (
+                time_since_last_sync is not None
+                and time_since_last_sync < self._excessive_sync_threshold
+            ):
                 logger.warning(
-                    "The previous sync of the experiment directory to the cloud "
-                    f"failed with the error: {str(e)}\nSyncing will be retried."
+                    "Experiment state snapshotting has been triggered multiple "
+                    f"times in the last {self._excessive_sync_threshold} seconds. "
```

Comment:
Also, why not suggest increase the `sync_up` period? instead of reduce the warning period.

Reply:
If `force=False`, the checkpointing period is `max(10, auto adjusted period)`, so we shouldn't be running into excessive syncs in the default case.

So, this message is just for the forced case.

Reply:
> why not suggest increase the sync_up period? instead of reduce the warning period?

`num_to_keep` will always cause experiment snapshots to be forced, which disregards the checkpoint period. So increasing the value of `TUNE_GLOBAL_CHECKPOINT_S` doesn't actually do anything. So, the only fixes are to increase `num_to_keep` or just accept this and suppress the warning. 🙁